### PR TITLE
search: server-side sort options

### PR DESCRIPTION
**Pull request by harunurhan**
* Sort option now comes with search results and rendered on the UI
dynamically, so that we can avoid hardcoding them on the UI and
creating different SortBy for each collections since sort options
change by collection.

* Makes [web] and [worker] services "restart: always" to avoid crashing
due to syntax errors during development

* Fixes literature record in `data/records` which had duplicate pid
(arxiv)

### CODE REVIEW

File: inspirehep/config.py [hunk 1]
```diff
@@ -422,14 +399,10 @@
                     "acronyms",
                     "titles",
                     "address",
-                    "opening_date"
-                    "cnum"
-                    "control_number",
+                    "opening_date" "cnum" "control_number",
```
Comment (drjova) [inline inspirehep/config.py]: ``,``

File: tests/integration/records/serializers/test_json.py [hunk 2]
```diff
@@ -314,6 +314,28 @@ def test_literature_json_ui_v1_response_search(api_client, db, create_record):
...
+    response = api_client.get("/literature", headers=headers)
+
+    response_status_code = response.status_code
+    response_data = json.loads(response.data)
+    sort_options = response_data["sort_options"]
+
+    assert expected_status_code == response_status_code
+    assert expected_sort_options == sort_options
+
+
```
Comment (drjova) [inline tests/integration/records/serializers/test_json.py]: it needs another test when is ``None``

File: ui/src/common/components/SortBy.jsx [hunk 3]
```diff
@@ -2,38 +2,30 @@ import React, { Component } from 'react';
...
     return (
-      <SelectBox
-        onChange={onSortChange}
-        defaultValue={sort}
-        options={SORT_BY_OPTIONS}
-      />
+      sortOptions && (
+        <SelectBox
+          onChange={onSortChange}
+          defaultValue={sort || sortOptions[0].value}
```
Comment (drjova) [inline ui/src/common/components/SortBy.jsx]: doesn't it make sense to specify which one is the default one in the config?

File: tests/integration/records/serializers/test_json.py [hunk 4]
```diff
@@ -622,6 +644,24 @@ def test_authors_default_json_v1_response_search(
...
+    expected_status_code = 200
+    expected_sort_options = None
+    response = api_client.get("/authors", headers=headers)
+
+    response_status_code = response.status_code
+    response_data = json.loads(response.data)
+    sort_options = response_data["sort_options"]
+
+    assert expected_status_code == response_status_code
+    assert expected_sort_options == sort_options
```
Comment (drjova) [inline tests/integration/records/serializers/test_json.py]: == ``None`` on the json response? :/
